woocommerce/pinterest-for-woocommerce
# Pinterest v5 user onboarding integration.

### Changes proposed in this Pull Request:



Closes #757 .

Integrating Pinterest extension onboarding process with Pinterest API v5. Porting API endpoints from elder versions to v5, changing onboarding process for the plugin.

### Detailed test instructions:

0. Disconnect from Pinterest account or make a fresh installation of Pinterest extension.
1. Checkout `pinterest-v5-integration-user-connect-wizard-integration` branch.
2. Follow the connection procedure.
3. You should successfully connect to Pinterest.

Being connected you could also try and sync Automatic Enhanced Match support. In order to test that, go to your Pinterest dashboard, find the tag and turn the switch on/off. Syncing setting inside the plugin should reflect the changes made at Pinterest side.

Disable the setting:

<img width="1392" alt="_2__Pinterest" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/7fbb685c-5797-4b9c-a255-fba1af145e9c">

Sync and check the extension got it synchronised.

<img width="1392" alt="Settings_‹_Pinterest_‹_Marketing_‹_WooCommerce_‹_WordPress_Pinterest_—_WooCommerce" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/f44ff160-6d2f-4e0d-aa87-3c19defdaf8f">

Being connected you could check auto tag creation for ad accounts which have no tag.
- create ad account

<img width="1392" alt="_2__Pinterest" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/4b1213c7-72a2-4942-99a5-8844045ae331">

- go to plugin connection settings and pick an ad account which has no tag from the dropdown. 

<img width="1469" alt="Connection_‹_Pinterest_‹_Marketing_‹_WooCommerce_‹_WordPress_Pinterest_—_WooCommerce" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/e80405db-558b-4ca1-83b6-2641c1254e60">

- observe a tag with a default name got created.

<img width="1039" alt="Cursor_and_Connection_‹_Pinterest_‹_Marketing_‹_WooCommerce_‹_WordPress_Pinterest_—_WooCommerce" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/29d2ee9b-5b29-4f2b-ae4f-d8a4d928b2ad">

4. Disconnect the extension from Pinterest.


### Changelog entry

> Upgrade - Pinterest API v5 connection process integration.
> Upgrade - Pinterest API v5 Settings and Connection tabs integration.
